tinygrad/tensor.py

```diff
@@ -552,6 +549,16 @@ def cast(self, dtype:DType) -> Tensor: return mlops.Cast.apply(self, dtype=dtype
   def float(self) -> Tensor: return self.cast(dtypes.float32)
   def half(self) -> Tensor: return self.cast(dtypes.float16)
 
+
+  # ***** Convenience stuff *****
+  @property
+  def ndim(self) -> int: return len(self.shape)
+  def numel(self) -> int: return math.prod(self.shape)
+  def element_size(self) -> int: return self.dtype.itemsize
+  def nbytes(self) -> int: return self.numel() * self.dtype.itemsize
+  def is_floating_point(self) -> bool: return dtypes.is_float(self.dtype)
```

Comment:
If you really want to keep this, at least remove the 8 lines of (untested) logic from dtypes. `self.dtype in [dtypes.float16, dtypes.float32]`

Reply:
Do you mean `is_int()`, `is_insgned()`? They are used in codegen.

How do you feel about 
```
class Dtype(...):
    [...]
    is_int: bool
    is_float: bool
    is_signed: bool
```

And then just `tensot.dtype.is_float`, etc in the code? This can turn 6 lines into 3.